src/main/cpp/src/api/genomicsdb.cc

```diff
@@ -0,0 +1,539 @@
+}
+
+
+GenomicsDB::GenomicsDB(const std::string& workspace,
+                       const std::string& callset_mapping_file,
+                       const std::string& vid_mapping_file,
+                       const std::string& reference_genome,
+                       const std::vector<std::string>attributes,
+                       const uint64_t segment_size) {
+  check(workspace, segment_size, callset_mapping_file, vid_mapping_file, reference_genome);
+    
+  // Create base query configuration
+  m_query_config = new VariantQueryConfig();
+
+  VariantQueryConfig* query_config = TO_VARIANT_QUERY_CONFIG(m_query_config);
+
+  query_config->set_workspace(workspace);
+  query_config->set_callset_mapping_file(callset_mapping_file);
+  query_config->set_vid_mapping_file(vid_mapping_file);
+  query_config->set_reference_genome(reference_genome);
+  query_config->set_attributes_to_query(attributes);
+
+  // Create storage manager
+  m_storage_manager = new VariantStorageManager(workspace, segment_size);
+
+  // Read in vid mapping information
+  m_vid_mapper = new FileBasedVidMapper(vid_mapping_file);
+}
+
+GenomicsDB::GenomicsDB(const std::string& query_configuration_json_file,
+                       const std::string& loader_configuration_json_file,
+                       const int concurrency_rank) : m_concurrency_rank(concurrency_rank) {
+  VERIFY(!query_configuration_json_file.empty() && "Specified query configuration file cannot be empty");
+
+  // Create base query configuration
+  m_query_config = new VariantQueryConfig();
+
+  VariantQueryConfig* query_config = TO_VARIANT_QUERY_CONFIG(m_query_config);
+
+  GenomicsDBImportConfig loader_config;
+  if (!loader_configuration_json_file.empty()) {
+    loader_config.read_from_file(loader_configuration_json_file, concurrency_rank);
+    query_config->update_from_loader(loader_config, concurrency_rank);
+  }
+
+  query_config->read_from_file(query_configuration_json_file, concurrency_rank);
+
+  check(query_config->get_workspace(concurrency_rank),
+        query_config->get_segment_size(),
+        query_config->get_reference_genome());
+
+  // Discard intervals not part of this partition
+  query_config->subset_query_column_ranges_based_on_partition(loader_config, concurrency_rank);
+
+  // Create storage manager
+  m_storage_manager = new VariantStorageManager(query_config->get_workspace(concurrency_rank), query_config->get_segment_size());
+}
+
+GenomicsDB::~GenomicsDB() {
+  // TODO: delete variant_query_config per array
+  if (m_vid_mapper != nullptr) {
+    delete TO_VID_MAPPER(m_vid_mapper);
+  }
+  if (m_query_config != nullptr) {
+    delete TO_VARIANT_QUERY_CONFIG(m_query_config);
+  }
+  if (m_storage_manager != nullptr) {
+    delete TO_VARIANT_STORAGE_MANAGER(m_storage_manager);
+  }
+}
+
+GenomicsDBVariants GenomicsDB::query_variants(const std::string& array,
+					       genomicsdb_ranges_t column_ranges,
+					       genomicsdb_ranges_t row_ranges) {
```

Comment:
should we also offer a way to pass in query_filter through this route? Or just ask the user to go the json route for that?

Reply:
Good idea, this was a first pass and I wanted to keep it simple. Just wondering if there is anything else needed for configuring these queries.